app/_components/SubmitGuessButton.tsx

```diff
@@ -110,32 +68,29 @@ function SubmitGuessButton({
           body: JSON.stringify(data),
         }
       );
-      // checking the tracking pin
-      for (const pin of pins) {
-        if (pin.poi_id === trackingPin?.poi_id) {
-          trackingPin.is_completed = true;
-        }
-      }
-      console.log("response", response);
+      const { message } = await response.json() as {message: string};
+      console.log(message);
     } catch (error) {
-      console.error("Error", error);
+      console.error(error);
     }
-  };
+  }
 
-  const handleSubmitGuessOnClick = () => {
-    navigator.geolocation.getCurrentPosition(
-      (position) => {
-        void evaluateGuess(position);
-      },
-      (error) => {
-        console.error("Error getting location:", error);
-      },
-      {
-        enableHighAccuracy: true,
-        timeout: 5000,
-        maximumAge: 0,
+  const handleSubmitGuessOnClick = async () => {
+    try {
+      if (!trackingPin) throw 'No pin to track';
+      if (!userCoordinates) throw 'No user coordinates'
```

Comment:
Can we make this function purer? I.e., it takes in `trackingPin` and `userCoordinates` as parameters instead of looking at them as global variables.